cql3/restrictions/statement_restrictions.cc

```diff
@@ -1357,6 +1477,23 @@ void statement_restrictions::prepare_indexed(const schema& idx_tbl_schema, bool
     }
     // If we're here, it means the index cannot be on a partition column: process_partition_key_restrictions()
     // avoids indexing when _partition_range_is_simple.  See _idx_tbl_ck_prefix blurb for its composition.
+
+    const column_definition* token_column = &idx_tbl_schema.clustering_column_at(0);
+
+    // In old indexes the token column was of type blob.
+    // This causes problems with sorting and must be handled separately.
+    _uses_v1_index = token_column->type != long_type;
```

Comment:
It's a bit ugly to have such a variable `_uses_v1_index` only needed for some outdated indexes that never get use nowadays. Why do we need this variable? It seems the code which uses it anyway has access to `idx_tbl_schema.clustering_column_at(0)` - so it can trivially check it type when it needs to, we don't need an extra variable.

Reply:
It's nicer to check things at prepare time rather than run time.

Reply:
Maybe I'm missing something, but it doesn't seem to be checking anything - it just keeps the result of a trivial calculation in a variable.

Reply:
The calculation is less trivial if you have to calculate token_column too.

Reply:
`token_column` has to be calculated anyway when calling `get_index_v1_token_range_clustering_bounds`.
 I will remove `_uses_v1_index`.